templates/tests/delete-test-templates.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+set -euo pipefail
+
+# Default to user HOME directory if not specified
+FOLDER_PATH=$HOME
+# Default to 'main' if not specified
+BRANCH_NAME="main"
+# Default to current logged in user if not specified
+ENV_NAME_PREFIX=$(whoami)
+TEMPLATE_NAME=""
+LOCATION="eastus2"
+# Default to a random value if not specified
+ENV_SUFFIX=""
+
+function usage {
+    echo "Deletes azd template"
+    echo ""
+    echo "Usage: delete-test-templates -t <template> -b <branch> -e <env_prefix>" 2>&1
+    echo ""
+    echo "  -s    Sets the environment suffix (default: RANDOM)"
+    echo "  -e    Sets the environment name prefix. Environment prefix is used in the azd environment name along with the template name (default: whoami)"
+    echo "  -f    Sets the root folder on the local machine for the test projects to be generated (default: User's HOME folder)"
+    echo "  -b    Sets the template branch name. Override to test a any custom branches (default: main)"
+    echo "  -t    Sets the template name. Use values from 'azd template list'. When omitted will run for all templates available in 'azd template list'"
+    echo "  -l    Sets the Azure location for the template infrastructure (default: eastus2)"
+    echo ""
+    echo "Examples:"
+    echo "  Deleting all templates with default values (must have environment suffix)"
+    echo "      bash ./delete-test-templates.sh -s \"<env_suffix>\""
+    echo ""
+    echo "  Deleting a single template with custom branch"
+    echo "      bash ./delete-test-templates.sh -t \"Azure-Samples/todo-nodejs-mongo\" -b \"<custom_branch_name>\" -s \"<env_suffix>\""
+
+    exit 1
+}
+
+while getopts "f:t:b:e:l:s:h" arg; do
+    case ${arg} in
+    f) FOLDER_PATH=$OPTARG ;;
+    t) TEMPLATE_NAME=$OPTARG ;;
+    b) BRANCH_NAME=$OPTARG ;;
+    e) ENV_NAME_PREFIX=$OPTARG ;;
+    l) LOCATION=$OPTARG ;;
+    s) ENV_SUFFIX=$OPTARG ;;
+    h)
+        usage
+        ;;
+    :)
+        echo "$0: Must supply an argument to -$arg." >&2
+        exit 1
+        ;;
+    ?)
+        echo "Invalid option -$arg."
+        exit 2
+        ;;
+    *) usage ;;
+    esac
+done
+
+# Cleans the specified template
+# $1 - The template name
```

Comment:
Since we don't use `$1` or `$2`, I would remove them and just keep `$3` (which will become `$1` but still means the environment name.)